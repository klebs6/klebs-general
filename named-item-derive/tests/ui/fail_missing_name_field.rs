// ---------------- [ File: tests/ui/fail_missing_name_field.rs ]
// named-item-derive/tests/trybuild/fail_missing_name_field.rs

#![allow(unused)]
use named_item_derive::NamedItem;
use named_item::*; // Named, etc.

#[derive(NamedItem)]
pub struct BadStruct {
    // We do not define `name: String`
    // => should fail with "Struct must have `name: String`."
    foo: i32,
}

fn main() {}
