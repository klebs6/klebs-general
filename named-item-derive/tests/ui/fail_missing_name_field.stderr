error: Struct must have `name: String`.
 --> tests/ui/fail_missing_name_field.rs:8:12
  |
8 | pub struct BadStruct {
  |            ^^^^^^^^^
